### PR TITLE
Add the missing break within the while statement of the gitea action

### PR DESCRIPTION
**Pull request by cmoulliard**
- Add the missing break within the while statement of the gitea action to exit when the gitea repository exists. 
- Issue: #23049

## Hey, I just made a Pull Request!

I added a break which was missing within this function which is checking if a gitea repository exists using the `gitea` action
```
async function checkAvailabilityGiteaRepository(
  maxDuration: number,
  integrationConfig: GiteaIntegrationConfig,
  options: {
    owner?: string;
    repo: string;
    defaultBranch: string;
    ctx: ActionContext<any>;
  },
) {
  const startTimestamp = Date.now();

  const { owner, repo, defaultBranch, ctx } = options;
  const sleep = (ms: number | undefined) => new Promise(r => setTimeout(r, ms));
  let response: Response;

  while (Date.now() - startTimestamp < maxDuration) {
    if (ctx.signal?.aborted) return;

    response = await checkGiteaContentUrl(integrationConfig, {
      owner,
      repo,
      defaultBranch,
    });

    if (response.status !== 200) {
      // Repository is not yet available/accessible ...
      await sleep(1000);
    } else {
      // Gitea repository exists !
      break;
    }
  }
}
```

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by cmoulliard** [comment]
Test result using code change 

Screenshot 2024-02-19 at 12 14 44
Screenshot 2024-02-19 at 12 14 57

**Review by vinzscam — APPROVED** [review]
makes sense 😄

File: qshift/templates/quarkus-application/skeletons/hello-world/pom.xml
Comment (benjdlambert) [inline qshift/templates/quarkus-application/skeletons/hello-world/pom.xml:1]: don't need this file here 👍

File: .changeset/weak-flowers-repeat.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-scaffolder-backend-module-gitea': patch
+---
+
+Fix issue for infinite loop when repository already exists
+Log the root cause of error reported by checkGiteaOrg
```
Comment (benjdlambert) [inline .changeset/weak-flowers-repeat.md:6]:
```suggestion
- Fix issue for infinite loop when repository already exists
- Log the root cause of error reported by `checkGiteaOrg`
```

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Just a few last bits 👍

**Comment by cmoulliard** [comment]
> Just a few last bits

Done

**Review by benjdlambert — APPROVED** [review]
Nice! Thanks for fixing everything up! :tada:

**Pull request merged**